### ISSUE TITLE
Mastodon/ActivityPub Bridge

### ISSUE THREAD

**Issue by gkbrk** [mdom/we-are-twtxt#17]
I am using a Mastodon bridge for my own twtxt account. The script seems flexible enough to be used with other ActivityPub services such as GNU social. It is currently a one-way bridge but it could be useful to follow accounts. Should important accounts be added to bots/users/a whole different file? Or will this simply be too noisy?

Here's a demo link: http://server.gkbrk.com/cgi-bin/twtxt.py?user=leo@niu.moe

**Comment by kseistrup** [comment]
Which ‘important accounts’? Twtxt? Mastodon? And what and who defines ‘important’ – and who?

**Comment by kseistrup** [comment]
If your aim is to display Mastodon streams as twtxt I think it's a bad idea: All those @-mentions will look like they're mentioning people on twtxt, when in fact they're just Mastodon users. It's going to bring confusion rather than value.

Is there a reason your own bridged twtxt file is shown in the opposite order of usual twtxt files?

**Comment by gkbrk** [comment]
My file is recent-first because that's how RSS and ActivityPub feeds look like, I can reverse it if it breaks people's scripts. When I was looking at random twtxt files in the wild I saw them ordered both ways.

The reason I asked about this topic is most project are starting to talk to each other. For example;
* Mastodon, GNU Social and PeerTube instances talk to each other in a federated fashion using ActivityPub
* Matrix servers can federate with each other. This includes people on public servers and people who join with their own personal server. Matrix can also talk to IRC and Slack using bridges.
* RSS and Email can be bridged with each other. RSS can also be bridged to ActivityPub and possibly twtxt formats.

So in short, open source and decentralized social media projects are able to talk to each other using common protocols. This increases adoption and gives people more choice about which software they can use.

I think bridging twtxt with other open formats is a good idea given the way other projects have been converging on common protocols.

The issue with @-mentions can be worked around by dynamically converting Mastodon mentions to Mastodon-bridge URLs. But in either case, this is just a question looking for discussion about this topic and not a demand by any means.

**Comment by kseistrup** [comment]
Twtxt files are usually created by appending new lines to an already existing file. If you see reversed files they're most likely created by converting data from another source into a twtxt format. It doesn't break anything, it just makes it obvious that it's not a usual stream.

I'm not keen on the idea of bridging twtxt, but this is not my repo.

**Comment by mdom** [comment]
I would be fine with bridges for bots, but in my opinion seeing just one side of a discussion on another network is a really bad user experience. This repository was meant as an easy to subscribe list of live streams on twtxt, so bridge accounts are probably not a good fit.
